### PR TITLE
SONARJAVA-1018 Rule: S2325 "private" methods that don't access instance data should be "static"

### PR DESCRIPTION
**Pull request by mpaladin**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/StaticMethodCheck.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  private boolean referenceInstance(Symbol symbol) {
+    return symbol.isUnknown() || (!symbol.isStatic() && fromInstance(symbol.owner()));
+  }
+
+  private boolean fromInstance(Symbol owner) {
+    for (Symbol outerClass : outerClasses) {
+      Type ownerType = owner.type();
+      if (owner.equals(outerClass) || (ownerType != null && outerClass.type().isSubtypeOf(ownerType))) {
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/StaticMethodCheck.java]: Tthe type of the owner should never be null. Did you encounter one?

Comment (mpaladin) [inline java-checks/src/main/java/org/sonar/java/checks/StaticMethodCheck.java]: @m-g-sonar that branch is covered by unit test.

File: java-checks/src/test/files/checks/StaticMethodCheck.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package foo;
+
+class Utilities {
```
Comment (m-g-sonar) [inline java-checks/src/test/files/checks/StaticMethodCheck.java]: could you add examples with usages of `super` and `this` to also show that these cases  are covered?

**Comment by m-g-sonar** [comment]
If you created a ticket for cases where owner is null, LGTM!

**Comment by mpaladin** [comment]
@m-g-sonar ticket created: http://jira.sonarsource.com/browse/SONARJAVA-1100.

**Pull request closed without merging**